### PR TITLE
Fixed GPU execution for dataset_api runtime test.

### PR DESCRIPTION
**Pull request by mhong**
All runtime tests are expected to pass in GPU mode now (though some control flow tests are still disabled for GPU and need to be debugged.) 

```
../llvm/utils/lit/lit.py -sv --param swift_tensorflow_gpu=true --param swift_test_mode=optimize --param swift_tensorflow_path=../build/bazel-bin/tensorflow ../build/Ninja-ReleaseAssert+stdlib-Release/swift-linux-x86_64/test-linux-x86_64/TensorFlowRuntime/
```

### CODE REVIEW

**Comment by mhong** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
Why not add it to `makeIterator()` also?

**Comment by mhong** [comment]
Overall we just need to annotate acccelerator-only functions (in this case it's the dataset creation function), which have to be on single-device.

The ops created by `makeIterator()` have GPU kernels (we may need to adjust the code for TPU later, where we can extend `DeviceTypeMgr::chooseDevice` for those ops to avoid placing them on TPU; that hack will be removed when we have integration with op device availability info from @jtkeeling): 

```
REGISTER_KERNEL_BUILDER(
    Name("MakeIterator").Device(DEVICE_GPU).HostMemory("dataset"),
    MakeIteratorOp);
REGISTER_KERNEL_BUILDER(Name("AnonymousIterator").Device(DEVICE_GPU),
                        AnonymousIteratorHandleOp);
```

**Comment by rxwei** [comment]
Hmm but the functions that you added the annotation to are not accelerator-only functions. `init(elements:elementShape:)` is the creation function as you said. `map(_:)` and `filter(_:)` are dataset operations. Neither of these are `@convention(tensorflow)`. So perhaps the definition should be: we need to annotate dataset operations that do not have a non-CPU kernel?

File: stdlib/public/TensorFlow/Dataset.swift [hunk 1]
```diff
@@ -46,6 +46,8 @@ public extension SingleValueDataset {
   // users must specify the element shape for today.
   @inlinable @inline(__always)
   public init(elements: Tensor<Scalar>, elementShape: TensorShape) {
+    // A dataset graph function must run on TF CPU.
+    TensorFlow.enableCPU()
```
Comment (rxwei) [inline stdlib/public/TensorFlow/Dataset.swift]: There's no need to say `TensorFlow.`. `enableXXX()` are just global functions.

**Comment by mhong** [comment]
You are right. We are not using `OneShotIterator`, which takes a function-typed attr `dataset_factory`, where the graph function needs to be accelerator-only. So in this case this is indeed about ops that do not have non-CPU kernels.

File: stdlib/public/TensorFlow/Dataset.swift (hunk 1 above)
Comment (mhong) [reply stdlib/public/TensorFlow/Dataset.swift]: Good catch. Removed.

**Review by rxwei — APPROVED** [review]
LGTM. Thanks!

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by mhong** [comment]
CI passed: https://ci-external.swift.org/view/Pull%20Request/job/swift-PR-TensorFlow-Linux/655/.

Thanks for the review!

**Pull request merged**